### DISCUSSION TITLE
how to configure cache by namespace per page (ssr mode)?

### DISCUSSION THREAD

**Discussion by academici** [Akryum/vue-apollo#767] [Q&A]
If use apollo InMemoryCache, service created common data storage.

Example:
Page1 has apollo query1, page2 - query2.
Go to page1, page contains data query1
Go to page2, page contains data query1 and query2

This is my vue apollo ssr initialize
```js
import "isomorphic-fetch";
import Vue from "vue";
import ApolloSSR from "vue-apollo/ssr";

if (process.server) {
  Vue.use(ApolloSSR);
}

const cacheKey = route => {
  let keyResult = `apollo-page-${route.name}`;
  for (let [key, value] of Object.entries(route.params)) {
    keyResult += `-${key}:${value}`;
  }
  return keyResult;
};

export default {
  async sendSSRData({ app, ssr, error }) {
    if (process.server) {
      try {
        ssr.bodyAdd = `<script>window.__APOLLO_STATE__=${JSON.stringify(
          ApolloSSR.getStates(app.$apolloProvider, {
            exportNamespace: cacheKey(app.$route)
          })
        )}</script>`;
      } catch (err) {
        error(err);
      }
    }
  }
};
```
Is it possible to configure cache adapter for page namespace key?
```js
// Custom starting link.
// If you want to replace the default HttpLink, set `defaultHttpLink` to false
link,
cache: new InMemoryCache()
...
```

Or is there some other solution?

**Comment by Austio** [comment]
@Akryum Think this is a good case to have some hooks that could handle this.  Something more granular that the global wrapper i pred here https://github.com/Akryum/vue-apollo/pull/718/files

**Comment by Akryum** [comment]
If you are doing SSR, you should have a new Vue app and new Apollo clients created for each page request.
